lib/ansible/playbook/__init__.py

```diff
@@ -61,6 +62,8 @@ def _load_playbook_data(self, file_name, variable_manager, vars=None):
         cur_basedir = self._loader.get_basedir()
         self._loader.set_basedir(self._basedir)
 
+        add_all_plugin_dirs(self._basedir)
```

Comment:
loader basedir might not be set on all paths, ansible-playbook does, but others might not

Reply:
This isn't using the loader basedir.  This is use `self._basedir` which is determined based on the `file_name` of the playbook.

Reply:
its more of an issue with plugins expectations, vars_plugins use loader to get data, if the directories don't match they might not get 'right config' or 'right data file'

Reply:
probably now true of inventory plugins also